source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -1,8 +1,236 @@
+   :language: go
+   :dedent:
+   :start-after: begin cursor next
+   :end-before: end cursor next
+
+.. note::
+   
+   If you previously used the cursor to iterate some documents, the
+   driver excludes those documents when you iterate again. 
+
+Documents Only in the Cursor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To attempt retrieving a document from your cursor, use the ``TryNext()``
+method.
+
+This method is unable to block the current goroutine. That means the
+function returns a document if each of the following is true:
+
+- A document is currently available
+- No errors occurred
+- The ``MongoClient`` didn't exhaust the cursor
+- The context didn't expire
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin cursor try next
+   :end-before: end cursor try next
+
+.. note::
+   
+   If you previously used the cursor to iterate some documents, the
+   driver excludes those documents when you iterate again. 
+
+.. _cursor-elements-golang:
+
+Access Cursor Elements
+----------------------
+
+You can access a cursors elements before it's **exhausted**. A cursor is
+exhausted when the ``MongoClient`` iterates through its results and
+reaches its last element. Afterwards, the cursor won't respond to
+functions to access its elements.
+
+When :ref:`individually retrieving your cursors results
+<individual-documents-golang>`, you can view each document as a ``BSON`` or
+Go value.
+
+BSON Document
+~~~~~~~~~~~~~
+
+To access the current document as raw ``BSON``, use the ``Current``
+property.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin current
+   :end-before: end current
+
+Go Value Document
+~~~~~~~~~~~~~~~~~
+
+To access the current document as a Go value, use the ``Decode()``
+function. 
+
+This function unmarshals the current document into the :ref:`BSON
+interface <bson-types>` you specify. It returns any errors from the
+unmarshalling process without any modification.
+
+.. code-block:: go
+
+   var result bson.D
+   if err := cursor.Decode(&result); err != nil {
+      log.Fatal(err)
+   }
+   fmt.Println(result)
+
+Other Elements
+~~~~~~~~~~~~~~
+
+You can also retrieve the following elements:
+
+- `The number of documents remaining in the batch. <{+api+}/mongo#Cursor.RemainingBatchLength>`__
+- `The cursor ID. <{+api+}/mongo#Cursor.ID>`__ 
+- `The cursor errors encountered. <{+api+}/mongo#Cursor.Err>`__
+
+.. important:: Retrieving the Cursor ID
+
+   If you want to retrieve the cursor ID, include the option to set your
+   batch size. The batch size needs to be a number less than or equal to
+   the number of documents you expect to retrieve.
+   
+   .. code-block:: go
+      
+      opts := options.Find().SetBatchSize(<a number>)
+      cursor, err := coll.Find(context.TODO(), bson.D{}, opts)
+      if err != nil {
+         panic(err)
+      }
```

Comment:
Suggestion: This is useful information, but I think should be omitted from this page as a discussion of why one would require the cursor ID is not included (I can't think of a use case off the top of my head for why one would need this outside of the Driver's source code, but maybe there is a use case).

Could a ticket be made to include this information in the API docs for the Driver? Could this note be removed from the page?